CMakeLists.txt

```diff
@@ -440,8 +440,12 @@ if(CURL_USE_OPENSSL)
   endif()
 
   set(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
-  check_symbol_exists(RAND_egd "${CURL_INCLUDES}" HAVE_RAND_EGD)
-  check_symbol_exists(OPENSSL_IS_BORINGSSL "openssl/base.h" HAVE_BORINGSSL)
+  if(NOT (HAVE_RAND_EGD))
+    check_symbol_exists(RAND_egd "${CURL_INCLUDES}" HAVE_RAND_EGD)
+  endif()
+  if(NOT (DEFINED HAVE_BORINGSSL))
```

Comment:
All the other cmake checks for HAVE_\* options use `if()`/`if(NOT)` without `DEFINED`. I think we should not introduce this inconsistency.

Reply:
I added an explanatory comment at the top.  I am checking for DEFINED rather than for the boolean value, because someone could define it to OFF, and in that case we still want to skip the symbol detection.